### PR TITLE
avoid unloading sched-simple

### PR DESCRIPTION
**Pull request by garlick**
These are sched changes needed in conjunction with flux-framework/flux-core#3177

### CODE REVIEW

**Comment by SteVwonder** [comment]
LGTM! A really nice cleanup @garlick!

**Comment by garlick** [comment]
Thanks @SteVwonder !   I was able to figure out the stuff that had me confused (mainly because i was testing against flux-framework/flux-core#3168 and there were other problems).

I'll update this in a sec and remove the WIP.  Since I found myself repeating a couple of code blocks over and over in tests, I added shell functions to `sched-sharness.sh` for loading hwloc XML into the core resource module, and for clearing active jobs before unloading the scheduler.  That should make the changes a bit easier to review.

File: t/sharness.d/sched-sharness.sh [hunk 1]
```diff
@@ -121,3 +121,22 @@ remove_qmanager () {
 remove_resource () {
     flux module remove sched-fluxion-resource "$@"
 }
+
+# Usage: reload_resources hwloc-dir
```
Comment (dongahn) [inline t/sharness.d/sched-sharness.sh:125]: Super minor nit.
```suggestion
# Usage: reload_resources hwloc-dir
# Usage: load_test_resources hwloc-dir
```

File: t/t1001-qmanager-basic.t [hunk 2]
```diff
@@ -10,6 +10,7 @@ excl_4N4B="${hwloc_basepath}/004N/exclusive/04-brokers"
 
 skip_all_unless_have jq
 
+export FLUX_SCHED_MODULE=none
```
Comment (dongahn) [inline t/t1001-qmanager-basic.t:13]: Typo in the commit message:

< use the load_test_resource()
> use the load_test_resources()

**Review by dongahn — APPROVED** [review]
Thanks @garlick. This not only makes fluxion ready with your upcoming system instance changes but also cleans up our tests!
 
1. There are couple of super minor nits in commit messages. 
2. The CI fails with test failures; and I assume that's because this PR waits for flux-framework/flux-core#3177 to land. 

1) is minor enough I approve.

File: t/t1001-qmanager-basic.t (hunk 2 above)
Comment (garlick) [reply t/t1001-qmanager-basic.t:13]: Thanks for catching those!  After we get flux-framework/flux-core#3177 merged (already approved), I'll force push those fixes and then we'll get another run of travis which should pass this time.

**Comment by garlick** [comment]
OK, flux-core merged the other PR, so I forced pushed the suggested changes here.  Let's see if travis likes this now.

**Comment by garlick** [comment]
Restarted one builder that seem to have run against the previous version of flux core (sched's configure reports `0.19.0-8-g9665e57dd`).

**Comment by garlick** [comment]
OK, that's got it. I'll go ahead and set MWP.

**Pull request merged**